CMakeLists.txt

```diff
@@ -1,9 +1,22 @@
-cmake_minimum_required(VERSION 3.12)
-project(SensemakerPythonLibrary)
+cmake_minimum_required(VERSION 3.20)
```

Comment:
Cmake's CUDA support in actively changes from version to version. By requiring a high Cmake version I was avoiding adding CUDA support differently depending on user's actual Cmake version. For example, earlier Cmake versions require CUDA_ADD_LIBRARY while later don't. Consider the following code example of what I was trying to avoid leaning towards "modern" Cmake:

if( ("${CMAKE_VERSION}" VERSION_EQUAL 3.8) OR ("${CMAKE_VERSION}" VERSION_GREATER 3.8) )
    # Modern CMake
    message(STATUS "---Modern Cmake---")
    if(USEGPU)
        enable_language("CUDA")
        set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
        set(CMAKE_CUDA_HOST_LINK_LAUNCHER ${CMAKE_CXX_COMPILER})
        add_executable(diana ${cpu_source_files})
        set_target_properties(nyxus PROPERTIES CUDA_ARCHITECTURES "35;37;50;72")
	target_link_libraries(nyxus ${Nyxus_LIBRARIES} -static-libgcc -static-libstdc++)
	add_library(nyxus_gpu ${gpu_source_files})
	target_link_libraries(nyxus nyxus_gpu)
    else(USEGPU)
        add_executable(nyxus ${cpu_source_files})
	target_link_libraries(${PROJECT_NAME} stdc++fs)
	target_link_libraries(${PROJECT_NAME} ${Nyxus_LIBRARIES})
    endif()
else()
    # Old CMake
    message(STATUS ">>>> Old Cmake")
    add_executable(diana ${cpu_source_files})
    if(USEGPU)
        find_package(CUDA 8.0)
        if(NOT CUDA_FOUND)
            message(STATUS "CUDA not found")
            set(USEGPU OFF)
        else()
            CUDA_ADD_LIBRARY(nyxus_gpu ${gpu_source_files})
            target_link_libraries(nyxus nyxus_gpu)
        endif()
    endif()
endif()